### PR TITLE
GPL-878-26 BUG - Pool/new page now clears all previous data when loaded

### LINKED ISSUE #658: GPL-878-26 [BUG] - On the new pool page the data isn't cleared when I navigate away or create a pool
**Describe the bug**

**RT Ticket Number**
If applicable

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to page '...'
2. Click on button '...'
3. See error

**Expected behaviour**
What was expected to happen.

**Screenshots**

**Desktop (please complete the following information):**
* Windows or Mac?
* Browser Chrome, Firefox, Safari or other?
* Browser version (use 'About' to look up)?

**Additional context**

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #658 

Changes proposed in this pull request:

* When navigating to pool/new it clears data from previous pools
* Footer cleaned up and fixed for e2e tests

### CODE REVIEW

File: src/components/InfoFooter.vue [hunk 1]
```diff
@@ -32,32 +32,32 @@ export default {
...
-        } catch (err) {
-          console.error(err)
-        }
-      } else {
-        this.repo = 'https://github.com/sanger/traction-ui/releases'
+      try {
+        await fetch('REPO')
+          .then((response) => response.text())
+          .then((response1) => {
+            // Checks returned text contains REPO text as fetch returns index.html if REPO doesn't exist
```
Comment (BenTopping) [inline src/components/InfoFooter.vue:54]: this is a bit crude but should do the trick, REPO is a file uploaded by the deployment scripts that contains the release being published

File: src/views/pacbio/PacbioPoolCreate.vue [hunk 2]
```diff
@@ -45,6 +45,8 @@ export default {
   created() {
     const plates = this.fetchPacbioPlates()
     const tagSets = this.fetchPacbioTagSets()
+    // Needed due to left over pool data from previously edited pools
+    this.$store.commit('traction/pacbio/poolCreate/clearPoolData')
```
[the remark this answers is not in the document]
Comment (BenTopping) [reply src/views/pacbio/PacbioPoolCreate.vue:49]: * dependent on response on other comment* , could add a mapMutations section to the file and call it in that way aswell

File: src/store/traction/pacbio/poolCreate/mutations.js [hunk 3]
```diff
@@ -110,4 +110,14 @@ export default {
...
   },
+  // This method clears the editable data in the pool/new page
+  clearPoolData: (state) => {
+    ;(state.libraries = {}),
+      (state.pool = {}),
+      (state.tube = {}),
+      (state.selected = {
+        tagSet: {},
+        plates: {},
+      })
```
[the remark this answers is not in the document]
Comment (JamesGlover) [reply src/store/traction/pacbio/poolCreate/mutations.js:121]: mapActions still uses the mutation, it just makes the syntax for calling a dispatch slightly cleaner:
https://vuex.vuejs.org/guide/actions.html#dispatching-actions-in-components

[the remark this answers is not in the document]
Comment (JamesGlover) [reply src/store/traction/pacbio/poolCreate/mutations.js:121]: Weird on the prettier though... can you try modifying it back to the something like the suggestion, and running prettier again. I'm wondering if something like a slight error in whitespace confused it.

Not sure if its worth disabling prettier is not, although it definitely is weird behaviour. @stevieing any thoughts?

[the remark this answers is not in the document]
Comment (BenTopping) [reply src/store/traction/pacbio/poolCreate/mutations.js:121]: ah yes found the issue, i had trailing commas after each line so prettier must have thought i was doing something weird with the object, after rerunning without commas its fine, my bad

[the remark this answers is not in the document]
Comment (JamesGlover) [reply src/store/traction/pacbio/poolCreate/mutations.js:121]: Ahh, as you pointed out in slack you're right. A mutation, not an action.

[the remark this answers is not in the document]
Comment (JamesGlover) [reply src/store/traction/pacbio/poolCreate/mutations.js:121]: It is possible to map mutations as well, but its less imporant where we only have one instance of it:
https://vuex.vuejs.org/guide/mutations.html#committing-mutations-in-components

**Pull request merged**